### DISCUSSION TITLE
TabbedContent – layout issue

### DISCUSSION THREAD

**Discussion by dragomirecky** [Textualize/textual#2822] [Q&A] [answered]
Hello!

First of all, amazing project 👏

I would like to start using it for some apps of mine but I quickly run into a layout issue. 

My minimal example
```python
from textual.app import App
from textual.widgets import TabbedContent, TabPane, Placeholder


class DiagnosticApp(App):
    def compose(self):
        with TabbedContent():
            with TabPane('PANE', id='pane'):
                yield Placeholder('pane content')


if __name__ == '__main__':
    app = DiagnosticApp()
    app.dark = False
    app.run()
```

Based on the examples on your site, I would expect to see the tab bar on the top and the rest of the screen filled with the "pane content" placeholder.

Instead, I see the tab bar only:
![Screenshot 2023-06-22 at 17 51 21](https://github.com/Textualize/textual/assets/1269664/a255b1c2-1ff8-402c-aaad-b87d41daa4d2)

Do you have any suggestions?

**Comment by davep** [comment]
Does the workaround mentioned to the bottom of https://github.com/Textualize/textual/issues/2411 work for you?

**Reply by dragomirecky** [reply]
It does not

Giving the `TabbedContent ContentSwitcher` background color, I can see that adding/removing `height: 1fr`; changes their size. But either way, the Placeholder isn't visible.

**Reply by willmcgugan** [reply]
Placeholders don't have a default size by default. Try the following:

```python
from textual.app import App
from textual.widgets import TabbedContent, TabPane, Placeholder


class DiagnosticApp(App):
    CSS = """
    Placeholder {
        height: 1;
    }
    """

    def compose(self):
        with TabbedContent():
            with TabPane("PANE", id="pane"):
                yield Placeholder("pane content")


if __name__ == "__main__":
    app = DiagnosticApp()
    app.dark = False
    app.run()
```

**Reply by dragomirecky** [reply]
Yep, that makes the Placeholder visible – height of one.

However, my goal is to make the children of the TabbedContent fill up the rest of the screen (in this minimal example the Placeholder; in reality – I want a ListView and some other things).

**Reply by willmcgugan** [reply, marked as answer]
```python
from textual.app import App
from textual.widgets import TabbedContent, TabPane, Placeholder


class DiagnosticApp(App):
    CSS = """
    TabbedContent ContentSwitcher{
        height: 1fr;
    }
    Placeholder {
        height: 1fr;
    }
    """

    def compose(self):
        with TabbedContent():
            with TabPane("PANE", id="pane"):
                yield Placeholder("pane content")


if __name__ == "__main__":
    app = DiagnosticApp()
    app.dark = False
    app.run()
```

**Reply by dragomirecky** [reply]
thanks 🤦‍♂️ that's it 👍

**Reply by dragomirecky** [reply]
and kudos for the super quick help 👏
